MDANSE/Src/MDANSE/Framework/AtomSelector/selector.py

```diff
@@ -13,362 +13,194 @@
+        function_parameters : Dict[str, Any], optional
+            the dictionary of keyword arguments defining a selection operation, by default {}
         """
-        if not self.check_valid_setting(settings):
-            raise ValueError(
-                f"Settings are not valid for the given chemical system - {settings}."
-            )
-
-        if reset_first:
-            self.reset_settings()
+        if number is None:
+            number = len(self.operations)
+        else:
+            try:
+                number = int(number)
+            except TypeError:
+                number = len(self.operations)
+        self.operations[number] = function_parameters
+
+    def validate_selection_string(
+        self, json_string: str, trajectory: Trajectory, current_selection: Set[int]
+    ) -> bool:
+        """Checks if the selection operation encoded in the input JSON string
+        will add any new atoms to the current selection on the given trajectory.
 
-        for k0, v0 in settings.items():
-            if isinstance(self.settings[k0], dict):
-                for k1, v1 in v0.items():
-                    self.settings[k0][k1] = v1
-            else:
-                self.settings[k0] = v0
-
-    def get_idxs(self) -> set[int]:
-        """The atom indices after applying the selection to the system.
+        Parameters
+        ----------
+        json_string : str
+            new selection operation in a JSON string
+        trajectory : Trajectory
+            a trajectory instance for which current_selection is defined
+        current_selection : Set[int]
+            set of currently selected atom indices
 
         Returns
         -------
-        set[int]
-            The atoms indices.
+        bool
+            True if the selection adds atoms, False otherwise
         """
-        idxs = set([])
-
-        for k, v in self.settings.items():
-            if isinstance(v, dict):
-                args = [{self._kwarg_keys[k]: i} for i in v.keys()]
-                switches = v.values()
+        function_parameters = json.loads(json_string)
+        if not self.operations:
+            return True
+        function_name = function_parameters.get("function_name", "select_all")
+        if function_name == "invert_selection":
+            selection = invert_selection(trajectory, current_selection)
+            return True
+        else:
+            operation_type = function_parameters.get("operation_type", "union")
+            function = function_lookup[function_name]
+            temp_selection = function(trajectory, **function_parameters)
+            if operation_type == "union":
+                selection = selection.union(temp_selection)
+            elif operation_type == "intersection":
+                selection = selection.intersection(temp_selection)
+            elif operation_type == "difference":
+                selection = selection.difference(temp_selection)
```

Comment:
Could be either 
```suggestion
            if operation_type == "union":
                selection |= temp_selection
            elif operation_type == "intersection":
                selection &= temp_selection
            elif operation_type == "difference":
                selection -= temp_selection
```
Or the shorter, but slightly more esoteric
```suggestion
            if operation_type in ("union", "intersection", "difference"):
                selection = getattr(selection, operation_type)(temp_selection)
```